core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -264,11 +268,12 @@ object ConsumerPerformance {
           val messageAndMetadata = iter.next
           messagesRead += 1
           bytesRead += messageAndMetadata.message.length
+          val currentTimeMillis = System.currentTimeMillis
 
-          if (messagesRead % config.reportingInterval == 0) {
+          if (currentTimeMillis - lastReportTime >= config.reportingInterval) {
```

Comment:
@junrao Thanks again for your feedback. To respond to your comments:
1) Sure, I'll update that for the new consumer. It seems to me the old consumer is OK.
2) Right, I see it happening for the new consumer only, and it's because we don't refresh the value of `currentTimeMillis` variable in case the `for` loop doesn't run. Actually, the fix for #1 above will fix this one too.
3) I'll try to run the system tests locally, but my resources are limited. In any case, @granders would you please give me access so I can run these tests on EC2? Thanks.